src/object_data.cpp

```diff
@@ -0,0 +1,46 @@
+#include "object_data.hpp"
+
+ObjectData::ObjectData() {
+    for (unsigned int i = 0; i < measurementDelay.size(); ++i) {
+        measurementDelay[i] = 0;
+    }
+}
+
+void ObjectData::addSample(Vector3D newPosition, uint32_t delayMs) {
+    for (unsigned int i = positions.size() - 1; i > 0; --i) {
+        positions[i] = positions[i - 1];
+        measurementDelay[i] = measurementDelay[i - 1];
+    }
+
+    positions[0] = newPosition;
+    measurementDelay[0] = delayMs;
+}
+
+Vector3D ObjectData::getPosition(const uint32_t &index) {
+    if (index < positions.size()) {
+        return positions[index];
+    }
+    return Vector3D();
+}
+
+Vector3D ObjectData::getSpeed(uint32_t index) {
+    return speed[index];
+}
+
+void ObjectData::setSpeed(const Vector3D &newSpeed) {
+    for (unsigned int i = speed.size() - 1; i > 0; --i) {
+        speed[i] = speed[i - 1];
+    }
+    speed[0] = newSpeed;
+}
+
+void ObjectData::clearSamples() {
```

Comment:
Empty method, please only declare methods without an implementation in the header. In this way, users who call the method are able to tell that there isn't an actual implementation.